notebooks/nessie-iceberg-hive-demo-nba.ipynb

```diff
@@ -0,0 +1,971 @@
+    "    \"\"\"\n",
+    "    Create a branch and switch the current cursor to the created branch\n",
+    "    \"\"\"\n",
+    "    hash_ = nessie_client.get_reference(nessie_client.get_default_branch()).hash_\n",
+    "    try:\n",
+    "        nessie_client.create_branch(ref, hash_)\n",
+    "    except:\n",
+    "        pass # already created\n",
+    "    \n",
+    "    return switch_ref_catalog(ref)\n",
+    "\n",
+    "\n",
+    "def switch_ref_catalog(ref):\n",
+    "    \"\"\"\n",
+    "    Switch a branch. When we switch the branch via Hive, we will need to reconnect to Hive\n",
+    "    \"\"\"\n",
+    "    # The important args below are:\n",
+    "    # catalog-impl: which Iceberg catalog to use, in this case we want NessieCatalog\n",
+    "    # uri: the location of the nessie server.\n",
+    "    # ref: the Nessie ref/branch we want to use (defaults to main)\n",
+    "    # warehouse: the location this catalog should store its data\n",
+    "    return hive.connect('localhost', configuration={'iceberg.catalog.dev_catalog.catalog-impl':'org.apache.iceberg.nessie.NessieCatalog',\n",
+    "                                                    'iceberg.catalog.dev_catalog.uri':'http://localhost:19120/api/v1',\n",
+    "                                                    'iceberg.catalog.dev_catalog.ref':ref,\n",
+    "                                                    'iceberg.catalog.dev_catalog.warehouse': warehouse}).cursor()\n",
+    "\n",
+    "print(\"\\n\\nHive running\\n\\n\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Solving Data Engineering problems with Nessie\n",
+    "============================\n",
+    "\n",
+    "In this Demo we are a data engineer working at a fictional sports analytics blog. In order for the authors to write articles they have to have access to the relevant data. They need to be able to retrieve data quickly and be able to create charts with it.\n",
+    "\n",
+    "We have been asked to collect and expose some information about basketball players. We have located some data sources and are now ready to start ingesting data into our data lakehouse. We will perform the ingestion steps on a Nessie branch to test and validate the data before exposing to the analysts."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Nessie branches (via Nessie CLI)\n",
+    "----------------------------\n",
+    "Once all dependencies are configured, we can get started with ingesting our basketball data into `Nessie` with the following steps:\n",
+    "\n",
+    "- Create a new branch named `dev`\n",
+    "- List all branches\n",
+    "\n",
+    "It is worth mentioning that we don't have to explicitly create a `main` branch, since it's the default branch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 316,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cursor = create_ref_catalog(\"dev\")"
```

Comment:
Is `cursor` a well known term in Hive or better use another term (maybe `ref` or `nessie_ref` or `current_branch` or something like that)?

Reply:
Not necessarily, it was just used by `pyHive` examples. I will change that